### PR TITLE
ROX-18114: Compliance by profile widget

### PR DESCRIPTION
**Pull request by bradr5**
## Description

Create the compliance by profile widget. Should display the percent of compliance by profile. Create a reusable component for both compliance by profile and compliance by cluster to use.

Note: currently uses static data

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed
Screenshot 2023-08-02 at 1 28 32 PM

### CODE REVIEW

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/HorizontalBarChart.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                    padding={{ bottom: 10 }}
+                    dependentAxis
+                />
+                <ChartGroup horizontal>
+                    {passingRateData.map(({ name, passing }) => (
+                        <ChartBar
+                            key={name}
+                            barWidth={defaultChartBarWidth}
+                            data={[{ x: name, y: passing }]}
+                            labels={({ datum }) => `${Math.round(parseInt(datum.y, 10))}%`}
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/HorizontalBarChart.tsx:61]: Has `parseInt` already truncated the value of `y` before `Math.round` gets its turn?

Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/HorizontalBarChart.tsx:61]: You're right. Depending on if the if `y` is a decimal I should either use parseFloat or just strip the Math.round portion. 

`Math.round(parseInt(datum.y, 10.6)) // 10`
`Math.round(parseFloat(datum.y, 10.6)) // 11`

but that brings up the question, how would we decide what to display for 99.9 or 0.1? Probably not super important to determine here but an interesting question nonetheless

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/types.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export type PassingRateData = {
+    name: string;
+    passing: number;
+    link: string;
+}[];
```
[the remark this answers is not in the document]
Comment (vjwilson) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/types.ts:5]: nit: I find it's better to specify singular types, and then use the array suffix `[]` on individual uses in the code, where you actually need a list of the thing.
```suggestion
export type PassingRateData = {
    name: string;
    passing: number;
    link: string;
};
```

[the remark this answers is not in the document]
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/types.ts]: Updated the typing. I do think there will be a lot of shared types at a level above once we introduce the other compliance sections but do you think it makes sense to throw all compliance types at the root of ComplianceEnhanced or does it make sense to have both root level and sub level typings?

[the remark this answers is not in the document]
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/types.ts]: Once API design is ready you components will import types for payloads and responses from requests from src/Services/ComplianceService2.ts

**Review by pedrottimark — APPROVED** [review]

**Pull request merged**